makeflow/src/makeflow_lambda_setup

```diff
@@ -0,0 +1,225 @@
+#!/bin/sh
+
+SCRIPT_NAME="$0"
+CONFIG_FILE=$1
+
+UUID="$(uuidgen)"
+
+ROLE_NAME="makeflow-role-$UUID"
+BUCKET_NAME="makeflow-bucket-$UUID"
+FUNCTION_NAME="makeflow-function-$UUID"
+
+FUNCTION_SOURCE="$FUNCTION_NAME".py
+FUNCTION_PACKAGE="$FUNCTION_NAME".zip
+ROLE_POLICY_FILE="makeflow-policy-$UUID.json"
+
+if [ $# -ne 1 ]
+then
+	echo "use: $0 <config-file>"
+	exit 1
+fi
+
+echo -n "Checking for aws command in PATH..."
+if which aws >/dev/null 2>&1
+then
+	echo "ok"
+else
+	echo "failed"
+	echo "$0: The \"aws\" command must be in your path to use this script."
+	exit 1
+fi
+
+echo -n "Checking for aws configuration..."
+if [ -f ~/.aws/config ]
+then
+	echo "ok"
+else
+	echo "failed"
+	echo "$0 You must run \"aws configure\" before using this script."
+	exit 1
+
+fi
+
+echo -n "Checking that aws command works..."
+if aws ec2 describe-instances > /dev/null 2>&1
+then
+	echo "ok"
+else
+	echo "failed"
+	echo "$0: Your Amazon credentials are not set up correctly. Try \"aws ec2 describe-instances\" to troubleshoot."
+	exit 1
+fi
+
+# Create the role that will be attributed to the Lambda function. This
+# gives the Lambda function 'permission' to access S3 and to execute
+
+echo -n "Creating role policy file $ROLE_POLICY_FILE..."
+
+# The skeleton of the role, to which we will attach the policies
+cat > "$ROLE_POLICY_FILE" <<- EOF
```

Comment:
The user needs permissions to create the roll and add it to themselves. Additionally, should the user already have these privileges before they can access the tools? That way if it's a student, the professor knows who has what permissions, and we're not adding powers to the users that they're not cleared to have.

Reply:
AFAIK, no special privileges are required b/c this is creating a fresh role unrelated to exisiting roles.  The user is simply granting to this function the ability to access this particular S3 bucket, so it's all self contained, I think.

Maybe see if you can run it yourself and run into any problems.

Reply:
Ah, I didn't read Conrad's code closely enough.  This creates a policy that let's the function access anything in S3 (that the user can access).  It should instead just grant access to a single bucket, e.g.:

https://aws.amazon.com/blogs/security/writing-iam-policies-how-to-grant-access-to-an-amazon-s3-bucket/